components/form.js

```diff
@@ -737,19 +810,85 @@ export function InputUserSuggest ({
             autoComplete='off'
             onChange={(formik, e) => {
               onChange && onChange(formik, e)
+              if (e.target.value === ovalue) {
+                // we don't need to set the ovalue or query if the value is the same
+                return
+              }
               setOValue(e.target.value)
-              setQuery(e.target.value.replace(/^[@ ]+|[ ]+$/g, ''))
+              setQuery(e.target.value.replace(prefixRegex, ''))
             }}
             overrideValue={ovalue}
             onKeyDown={onKeyDown}
             onBlur={() => setTimeout(resetSuggestions, 500)}
           />
         )}
-      </UserSuggest>
+      </SuggestComponent>
     </FormGroup>
   )
 }
 
+export function InputUserSuggest ({
+  transformUser, filterUsers, ...props
+}) {
+  return (
+    <BaseInputSuggest
+      transformItem={transformUser}
+      filterItems={filterUsers}
+      SuggestComponent={UserSuggest}
+      prefixRegex={/^[@ ]+|[ ]+$/g}
+      {...props}
+    />
+  )
+}
+
+export function InputTerritorySuggest ({
+  transformSub, filterSubs, ...props
+}) {
+  return (
+    <BaseInputSuggest
+      transformItem={transformSub}
+      filterItems={filterSubs}
+      SuggestComponent={TerritorySuggest}
+      prefixRegex={/^[~ ]+|[ ]+$/g}
+      {...props}
+    />
+  )
+}
+
+function UserSuggest ({
+  transformUser = user => user, filterUsers = () => true,
+  children, ...props
+}) {
+  return (
+    <BaseSuggest
+      transformItem={transformUser}
+      filterItems={filterUsers}
+      getSuggestionsQuery={USER_SUGGESTIONS}
+      itemsField='userSuggestions'
+      {...props}
+    >
+      {children}
+    </BaseSuggest>
+  )
+}
+
+function TerritorySuggest ({
+  transformSub = sub => sub, filterSubs = () => true,
+  children, ...props
+}) {
+  return (
+    <BaseSuggest
```

Comment:
The only change I made to these were removing all the parameters that weren't needed due to `...props`.